pacific-atlantic-water-flow/sonjh1217.swift

```diff
@@ -0,0 +1,118 @@
+class Solution {
+    // O(m*n) time / O(m*n) space
+    func pacificAtlanticBFS(_ heights: [[Int]]) -> [[Int]] {
+        var result = [[Int]]()
+        guard let first = heights.first else {
+            return result
+        }
+        let m = heights.count
+        let n = first.count
+        
+        var goesPacific = Array(repeating: Array(repeating: false, count: n), count: m)
+        var goesAtlantic = Array(repeating: Array(repeating: false, count: n), count: m)
+        let directions = [(1,0), (-1,0), (0,1), (0,-1)]
+        
+        func bfs(queue: [(Int, Int)], goes: inout [[Bool]]) {
+            var queue = queue
+            var head = 0
+            for (i, j) in queue {
+                goes[i][j] = true
+            }
+            
+            while head < queue.count {
+                var (i, j) = queue[head]
+                head += 1
+                
+                for (di, dj) in directions {
+                    let ni = i + di
+                    let nj = j + dj
+                    if ni >= 0
+                        && ni < m
+                        && nj >= 0
+                        && nj < n
+                        && !goes[ni][nj]
+                        && heights[ni][nj] >= heights[i][j]  {
+                        goes[ni][nj] = true
+                        queue.append((ni, nj))
+                    }
+                }
+            }
+        }
```

Comment:
재밌는 거 하나 공유하고 갑니다! (저만 재밌음 주의)
최단거리 구하지는 못하지만 Flood-Fill 하기는 좋아요!

```suggestion
        func depthFirstBreadthSecondSearch(queue: [(Int, Int)], goes: inout [[Bool]]) {
            var stack = queue
            var head = 0
            for (i, j) in stack {
                goes[i][j] = true
            }
            
            while !stack.isEmpty {
                let (i, j) = stack.removeLast()
                
                for (di, dj) in directions {
                    let ni = i + di
                    let nj = j + dj
                    if ni >= 0
                        && ni < m
                        && nj >= 0
                        && nj < n
                        && !goes[ni][nj]
                        && heights[ni][nj] >= heights[i][j]  {
                        goes[ni][nj] = true
                        stack.append((ni, nj))
                    }
                }
            }
        }
```

Reply:
어떤 점이 좋은가요?